File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void flatten(TreeNode* root) {
        if (!root) return;
            
        vector<TreeNode*> q;
        
        pre(root, q);
        
        for (int i = 0; i < q.size() - 1; i++) {
            q[i]->left = nullptr;
            q[i]->right = q[i + 1];
        }
    }
    
    void pre(TreeNode* root, vector<TreeNode*>& q) {
        if (!root)
            return;
        
        q.push_back(root);
        
        pre(root->left, q);
        pre(root->right, q);
    }
};

File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def flatten(self, root: Optional[TreeNode]) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        if not root: return
        
        q = []
        
        self.pre(root, q)
        
        for index in range(len(q) - 1):
            q[index].left = None
            q[index].right = q[index + 1]
        
    def pre(self, root, q):
        if not root: return
        
        q.append(root)
        
        self.pre(root.left, q)
        self.pre(root.right, q)